src/selections/selection-test.js

```diff
@@ -73,13 +73,15 @@ let linkSelectionTest = function(payload) {
 	);
 };
 
-let imageSelectionTest = function(payload) {
-	let selectionData = payload.data.selectionData;
+const imageSelectionTest = function(payload) {
+	const selectionData = payload.data.selectionData;
+	const element = selectionData.element;
+	const hasImage = !!element && !!element.findOne('img');
+	const isImage = !!element && element.getName() === 'img';
 
 	return !!(
-		selectionData.element &&
-		selectionData.element.getName() === 'img' &&
-		!selectionData.element.isReadOnly()
+		element &&
+		(hasImage || isImage)
```

Comment:
So something like this?

```javascript
const selectionData = payload.data.selectionData;
const element = selectionData.element;
const hasImage = element && element.findOne('img') !== null;
const isImage = element && element.getName() === 'img';

return !!(
  element &&
  (hasImage || isImage)
);
```

Reply:
Yeah. In general, my thinking on coercing to bool is that I only want to do it where the type checker tells me I have to do it, in order to write "idiomatic" JS. I know we're not using a type-checker, but as a thought experiment, you can image we'd annotate this function as having return type `boolean`, which means that we'd only need one `!!` (in the expression returned by the`return` statement).

https://www.typescriptlang.org/play/#src=function%20booly(a%3A%20string%2C%20b%3A%20string)%3A%20boolean%20%7B%0D%0A%20%20return%20!!(a%20%26%26%20b)%3B%0D%0A%7D

Reply:
I see - thanks for the context.
Here `selectionData.element` can be `null` that's why I was using these `!!` all over the place but they're not needed